File: Equal_0_1.py
binary_str = "101010110011"

seen_indices = {0: -1}
balance = 0
max_len = 0

for idx, char in enumerate(binary_str):
    balance += 1 if char == "1" else -1

    if balance in seen_indices:
        sub_len = idx - seen_indices[balance]
        max_len = max(max_len, sub_len)
    else:
        seen_indices[balance] = idx

print(max_len)

File: Equal_0_1.cpp
#include <iostream>
#include <string>
#include <unordered_map>
#include <algorithm>

using namespace std;

int main() {
    string binaryStr = "1001110011";
    
    unordered_map<int, int> firstSeen;
    firstSeen[0] = -1;
    
    int balance = 0;
    int maxLen = 0;

    for (int idx = 0; idx < binaryStr.length(); idx++) {
        balance += (binaryStr[idx] == '1') ? 1 : -1;

        int match = firstSeen.find(balance);
        if (match != firstSeen.end()) {
            int currentLen = idx - match->second;
            if (currentLen > maxLen) {
                maxLen = currentLen;
            }
        } else {
            firstSeen[balance] = idx;
        }
    }

    cout << maxLen << endl;

    return 0;
}
